W-14690733 MQ Cross Region Failover (#150)
mulesoft/docs-mq: modules/ROOT/pages/mq-apis.adoc

<.> The fallback region for this queue is `us-west-2`.
====
Fallback Queue::
+
====
[source,json]
----
[
{
    "encrypted" : true,
    "type" : "queue",
    "queueId" : "test_fb",
    "fifo" : false,
    ...
    "isFallback" : true,                     // <.>
    "fallbackConfig" : {
        "enabled" : true,                    // <.>
        "linkedQueueName" : "test",          // <.>
        "linkedQueueRegion" : "ca-central-1" // <.>
    }
},
]
----
<.> This queue is a fallback queue.
<.> Failover is enabled. If the primary region goes down, apps publish and consume from this queue.
<.> The primary queue that is linked to this fallback queue is `test`.
<.> The region for the primary queue that is linked to this fallback queue is `ca-central-1`.
====

Standard Queue Without Failover Configured::
+
Response includes `"isFallback" : false` and doesn't include `fallbackConfig`.
+
====
[source,json]
----
[
  {
  "encrypted" : true,
  "type" : "queue",
  "queueId" : "hello1",
  "fifo" : false,
  ...
  "isFallback" : false  // <.>
},
]
----
<.> This queue is a primary queue.
====

> FIFO Queue::

Reviewer: I'm inclined towards removing FIFO Queue and Exchange sections here because this API content is taking up too much space on the docs page. Since this feature is not supported for FIFO queues and exchanges, customers will most likely not be looking for fallback info for these destination types in the response. Let me know what you think :)

Author: Yeah, my thinking was to emphasize what's not supported. We can collapse them with twisties...

Reviewer: Collapsing them sounds good to me

Author: Done

* Message routing rule requirements, see
xref:mq-routing-rules.adoc#intelligent-routing-restrictions[Requirements and Restrictions].

[[enable-failover-api]]
=== Enable or Disable Failover on an Existing Standard Queue

When you enable failover for a queue, Anypoint MQ automatically creates a fallback queue in a xref:mq-failover.adoc#primary-fallback-regions[fallback region].
If a Mule app can't reach an Anypoint MQ server in a region,
it switches to the fallback queue to publish and consume messages.

> For information about situations that can cause a region to be considered unavailable, see xref:mq-failover.adoc#region-unavailability[Region Unavailability].

Reviewer 2: ```suggestion
For information about situations that can cause a region to be unavailable, see xref:mq-failover.adoc#region-unavailability[Region Unavailability].
```

Reviewer 2: Suggestion only. You may have a reason for using that word.

Author: Yeah, I want to leave it because it's not definitively unavailable.

        "linkedQueueName" : "00-test_fb", // <.>
        "linkedQueueRegion" : "us-west-2" // <.>
    }
},
]
----
<.> This queue is a primary queue.
<.> Failover is enabled for this queue. 
<.> The fallback queue for this queue is `00-test_fb`. If the primary region goes down, apps publish and consume from this queue.
<.> The fallback region for this queue is `us-west-2`.
====

.Fallback Queue
[%collapsible]
====
[source,json]
----
[
{
    "encrypted" : true,
    "type" : "queue",
    "queueId" : "test_fb",
    "fifo" : false,
    ...
    "isFallback" : true,                     // <.>
    "fallbackConfig" : {
        "enabled" : true,                    // <.>
        "linkedQueueName" : "test",          // <.>
        "linkedQueueRegion" : "ca-central-1" // <.>
    }
},
]
----
<.> This queue is a fallback queue.
<.> Failover is enabled. If the primary region goes down, apps publish and consume from this queue.
<.> The primary queue that is linked to this fallback queue is `test`.
<.> The region for the primary queue that is linked to this fallback queue is `ca-central-1`.
====

.Standard Queue Without Failover Configured
[%collapsible]
====

> Response includes `"isFallback" : false` and doesn't include `fallbackConfig`.

Reviewer 2: ```suggestion
The response includes `"isFallback" : false` and doesn't include `fallbackConfig`.
```

[source,json]
----
[
  {
  "encrypted" : true,
  "type" : "queue",
  "queueId" : "hello1",
  "fifo" : false,
  ...
  "isFallback" : false  // <.>
},
]
----
<.> This queue is a primary queue.
====

.FIFO Queue
[%collapsible]
====

> Response doesn't include `fallbackConfig` because FIFO queues don't support failover.

Reviewer 2: ```suggestion
The response doesn't include `fallbackConfig` because FIFO queues don't support failover.
```

[source,json]
----
{
  "encrypted" : true,
  "type" : "queue",
  "queueId" : "hello1",
  "fifo" : true,       // <.>
  ...
  "isFallback" : false
},
]
----
<.> This queue is a FIFO queue, which doesn't support failover. 
====

.Exchange
[%collapsible]
====

> Response doesn't include `fallbackConfig` because exchanges don't support failover.

Reviewer 2: ```suggestion
The response doesn't include `fallbackConfig` because exchanges don't support failover.
```

[[historical-stats-queue]]
=== Get Historical Statistics for a Queue

Use this API to fetch historical statistics for a queue.

> However, it does not get updated in real-time, and there is an approximate latency of 5 to 10 minutes before data is published. The retention period of historical data is 15 months.

Reviewer 2: ```suggestion
Statistics don't get updated in real-time, and there is an approximate latency of five to ten minutes before data is published. The retention period of historical data is 15 months.
```
